pkg/apply/cache/resource_cache_map.go

```diff
@@ -50,13 +56,19 @@ func (rc *ResourceCacheMap) Put(value *unstructured.Unstructured) error {
 
 // Set the resource in the cache using the supplied key, and replacing any
 // existing resource with the same key.
-func (rc *ResourceCacheMap) Set(key object.ObjMetadata, value *unstructured.Unstructured) {
+func (rc *ResourceCacheMap) Set(key object.ObjMetadata, value ResourceStatus) {
```

Comment:
Do we need this function, or can we compute the key based on the Unstructured in the ResourceStatus?

Reply:
That's what Put does. But in most cases we already have the ObjMetadata computed.